AirSkyBoat/AirSkyBoat
# [core] Fix NATION_CONTROL and ZONE_HOME_NATION latents.

**_I affirm:_**


## Please enter a player-facing description


This PR applies the following fixes:
- An issue where equipping gear with "areas under own nation's control", "in areas outside of own nation's control", and "in Nation" latent effects would cause the server to crash has been addressed.
- An issue where zoning or logging in with gear that uses "areas under own nation's control", "in areas outside of own nation's control", and "in <Nation>" latent effects would not properly trigger those latent effects has been addressed.

## What does this pull request do? (Please be technical)
The `conquest_system` table (and by extension, the `regionControls` array) only contains information on regions subject to the conquest system. When retrieving data on a region not subject to that system that belongs in the Vanilla/RotZ/CoP areas, the `getRegionOwner` function will access an index outside of the `regionControls` array, causing an out-of-bounds error which crashes the server. The proposed fix is to set a case for special regions (cities, Dynamis/Limbus, Lumoria/Sea) that are not subject to conquest. 

Additionally, there is a change to the logic regarding the activation of the latent when the user is either logging in or zoning into a region (not) under the control of the the player's nation. The same change is also done to the "Inside Nation" latent (as utilized on nation aketons). In addition to checking if the player is zoning via the `loc.zone` value, a check also done on the player's `loc.destination`. The case appears to be that `loc.destination` is set to a non-zero value whenever `loc.zone` is set to  a `nullptr`.


Closes #3383 

## Steps to test these changes
- "Areas under own nation's control" latent
  - !additem 14157 
    - (T.M. Pigaches +1)
- "Areas outside own nation's control"
  - !additem 14015
    - (Praefectus's Gloves)
- "In Nation"
  - !additem 14428 (Kingdom aketon)
  - !additem 14429 (Republic aketon)
  - !additem 14430 (Federation aketon)


Test by equipping the above items and travelling to and logging in to the following zones and checking your status, combat skills (T.M Pigaches +1/Praefectus's Gloves) and !getmod MOVE (Aketons):
- Any zone in Bastok
- Any zone in Windurst
- Any zone in San d'Oria
- Tavnazian Safehold
- Any zone in Jeuno 
- Any zone in Lumoria
  - Al'Taieu
  - The Garden of Ru'Hmet
  - Empyreal Paradox
- Any Promyvion zone.
- Any Apollyon/Temenos zone.

In all of these areas, the T.M Pigaches latent effect should *NEVER* activate and the Praefectus's Gloves latent effect should *ALWAYS* activate.

## Special Deployment Considerations
None.
